### PR TITLE
feat(RNPSW): :  add plan and invoice_limit props for pre-defined plan subscription

### PR DESCRIPTION
**Pull request by opmat**
add plan prop to provide an option for creating subscription to predefined plan on Paystack.

This is in response to [issue #162](https://github.com/just1and0/React-Native-Paystack-WebView/issues/162) and issue #128

### CODE REVIEW

**Comment by just1and0** [comment]
@opmat thanks for making these changes. We made some updates to the contribution docs. Could you please update the PR to follow the guidelines?

**Comment by opmat** [comment]
@just1and0 Thank you for pointing this out. The update has been made and I believe it is in line, as requested

File: development/paystack.tsx [hunk 1]
```diff
@@ -69,6 +71,11 @@ const Paystack: React.ForwardRefRenderFunction<React.ReactNode, PayStackProps> =
   const dynamicSplitString = dynamicSplitObjectIsValid(split) ? `split: ` + JSON.stringify(split)  + `,` : ''; // should only send split for dynamic multi-account split with the correct split object as defined
   //Sometimes, you can't determine a split configuration until later in the purchase flow. With dynamic splits, you can create splits on the fly. This can be achieved by passing a split object
 
+  const planCodeString = plan ? `ref: '${plan}',` : ''; // should only send plan with the predefined plan_code as generated on paystack dashboard if present, else if blank, it will be ignored.
```
Comment (just1and0) [inline development/paystack.tsx:74]: @opmat `refNumberString` already returns `ref:`, passing  `planCodeString` might not override `refNumberString` and end up causing problems. 

Will there be a case where refNumberString is not needed?

**Review by just1and0 — CHANGES_REQUESTED** [review]
Hey @opmat , Thanks for making these changes.  I left some more reviews.

File: development/paystack.tsx (hunk 1 above)
Comment (opmat) [reply development/paystack.tsx:74]: @just1and0 thank you for the quick catch. This has been corrected and pushing an updating now

**Comment by opmat** [comment]
@just1and0  thank you for the quick catch. This has been corrected and pushed an update now

**Review by just1and0 — APPROVED** [review]
LGTM

**Comment by just1and0** [comment]
@all-contributors add @opmat for code
@all-contributors add @opmat for docs

**Pull request merged**